test/wasm2js/start_func.2asm.js.opt

```diff
@@ -0,0 +1,61 @@
+
+function asmFunc(global, env, buffer) {
+ var HEAP8 = new global.Int8Array(buffer);
+ var HEAP16 = new global.Int16Array(buffer);
+ var HEAP32 = new global.Int32Array(buffer);
+ var HEAPU8 = new global.Uint8Array(buffer);
+ var HEAPU16 = new global.Uint16Array(buffer);
+ var HEAPU32 = new global.Uint32Array(buffer);
+ var HEAPF32 = new global.Float32Array(buffer);
+ var HEAPF64 = new global.Float64Array(buffer);
+ var Math_imul = global.Math.imul;
+ var Math_fround = global.Math.fround;
+ var Math_abs = global.Math.abs;
+ var Math_clz32 = global.Math.clz32;
+ var Math_min = global.Math.min;
+ var Math_max = global.Math.max;
+ var Math_floor = global.Math.floor;
+ var Math_ceil = global.Math.ceil;
+ var Math_sqrt = global.Math.sqrt;
+ var abort = env.abort;
+ var nan = global.NaN;
+ var infinity = global.Infinity;
+ function foo() {
+  HEAP32[0] = 2;
+ }
+ 
+ var FUNCTION_TABLE = [];
+ foo();
```

Comment:
Is it a problem that the call to the start function is emitted before the intrinsic functions below? What will happen if the start function contains a memory.size instruction?

Reply:
That is ok in JS - you can call a declared function (`function x() { }`) before the declaration is reached (unlike `var x = function(){};` or in python).

And memory.size would work because we create memory (and segments) before getting here.